src/Microsoft.TestPlatform.CommunicationUtilities/SocketCommunicationManager.cs

```diff
@@ -89,10 +95,12 @@ internal SocketCommunicationManager(IDataSerializer dataSerializer)
         /// <summary>
         /// Host TCP Socket Server and start listening
         /// </summary>
+        /// <param name="endPointAddress">End point where server is hosted</param>
         /// <returns>Port of the listener</returns>
-        public int HostServer()
+        public int HostServer(string endPointAddress)
         {
-            var endpoint = new IPEndPoint(IPAddress.Loopback, 0);
+            var endPoint = endPointAddress.Split(':');
```

Comment:
Will not work for IPv6? Should the API ask for IPEndpoint directly? All consumers are first doing a string concat, then we split it to create ip endpoint?